backend/src/main/java/gov/cdc/usds/simplereport/api/pxp/PatientExperienceController.java

```diff
@@ -92,7 +94,25 @@ public PxpVerifyResponse getPatientLinkVerify(
     return new PxpVerifyResponse(p, os, te);
   }
 
+  /**
+   * Given a patient link ID, returns the name of the patient for whom the test was conducted, in
+   * lightly-obfuscated form (e.g. "John Doe" -> "John D").
+   *
+   * <p>This endpoint is unauthorized and therefore does not return fully identifying data.
+   *
+   * @param request
+   * @return an obfuscated patient name
+   */
+  @GetMapping("/patient-name")
+  public String getObfuscatedPatientNameFromLink(
+      @RequestParam("patientLink") String patientLink, HttpServletRequest request) {
+    UUID link = UUID.fromString(patientLink);
```

Comment:
It may be worth adding logic here to check that the patient link is valid before we try to convert to a UUID. (I think that's done in the context holder for the endpoint above; we're trying to move away from those but perhaps we can have a helper method in this Controller that takes a patient link and verifies it.) 

I only mention this because we were paged several months ago for a UUID issue - patients were either removing bits off the end of the link or double-pasting it, so that the UUID wasn't valid, and that caused all kinds of cascading exceptions.

Reply:
That's a good call-out. Fortunately, the frontend has an early-out for malformed patient link IDs (https://github.com/CDCgov/prime-simplereport/blob/main/frontend/src/patientApp/PatientApp.tsx#L54-L70) so in common use we should be safe for that. That said, it doesn't hurt to have the API play defense against this too! See 0a8d147.

<img width="998" alt="Screen Shot 2022-02-02 at 2 33 48 PM" src="https://user-images.githubusercontent.com/27730981/152224855-204367e8-eac8-4e7a-ae5a-adb7cdfbc81f.png">

Reply:
Actually, I guess that commit doesn't really change anything... are you thinking of having this throw an `InvalidPatientLinkException` you mean?

Reply:
As @zdeveloper wisely points out, Spring gives us this for free if we simply type the incoming query param as `UUID` 🙃 

See [f52793c](https://github.com/CDCgov/prime-simplereport/pull/3305/commits/f52793cdc3fd07c763da6973a604588dd645e633)

<img width="908" alt="Screen Shot 2022-02-02 at 6 03 43 PM" src="https://user-images.githubusercontent.com/27730981/152252421-269694b7-3054-44d2-a121-6e8a07bd880d.png">

Reply:
Discussed offline; I suggested a `BadRequestException` to avoid triggering the 4xx error.

Reply:
Oh I only just saw your last comment Nathan, my apologies! Using the UUID type looks good 👍